server/routes/auth/slack.js

```diff
@@ -0,0 +1,49 @@
+const router = require('express').Router()
+const request = require('request')
+const { SLACK_CLIENT_ID, SLACK_CLIENT_SECRET } = require('../../../secrets')
+const { User } = require('../../db')
+module.exports = router
+
+router.get('/callback', function(req, res) {
+  if (!req.query.u) return res.status(400).send('No user')
+  if (!req.query.code) return res.status(400).send('No code')
+
+  const url =
+    'https://slack.com/api/oauth.access?client_id=' +
+    SLACK_CLIENT_ID +
+    '&client_secret=' +
+    SLACK_CLIENT_SECRET +
+    '&code=' +
+    req.query.code +
+    '&redirect_uri=http://localhost:3001/api/v0/auth/slack/callback?u=' +
```

Comment:
Same thing here as I said above with dropbox - how does the `redirect_uri` work? Aren't you already inside that route?